cmake/modules/Utility.cmake

```diff
@@ -206,19 +201,20 @@ function(add_swift_module target name deps sources additional_args)
   endforeach()
 
   # Add rpath to lookup the linked dylibs adjacent to itself.
-  if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
     list(APPEND DYLYB_ARGS -Xlinker -rpath -Xlinker @loader_path)
-    list(APPEND DYLYB_ARGS -Xlinker -install_name -Xlinker @rpath/${target}.${DYLIB_EXT})
-  else()
-    list(APPEND DYLYB_ARGS -Xlinker "-rpath=\\$$ORIGIN")
-  endif()
+    list(APPEND DYLYB_ARGS -Xlinker -install_name -Xlinker @rpath/${target}${CMAKE_SHARED_LIBRARY_SUFFIX})
 
-  # Runpath for finding Swift core libraries in the toolchain.
-  # FIXME: Ideally, this should be passed from the swift-ci invocation.
-  if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    # Runpath for finding Swift core libraries in the toolchain.
+    # FIXME: Ideally, this should be passed from the swift-ci invocation.
     list(APPEND DYLYB_ARGS -Xlinker -rpath -Xlinker @loader_path/../../macosx)
-  else()
+  elseif(CMAKE_SYSTEM_NAME STREQUAL Linux)
+    list(APPEND DYLYB_ARGS -Xlinker "-rpath=\\$$ORIGIN")
     list(APPEND DYLYB_ARGS -Xlinker "-rpath=\\$$ORIGIN/../../linux")
+  elseif(CMAKE_SYSTEM_NAME STREQUAL Windows)
```

Comment:
It's fine to not do this for this patch, but in future I'd put the Windows condition above the Linux one, turn Linux into the "else" case, and then construct the last path component of the rpath argument using the lowercased form of `uname -s` - that way Linux, FreeBSD, and any other Unix-like OSes fall into the default case automatically and get something sensible.

Reply:
That definitely wouldn't work here...the rpath encodes the OS name itself.

Reply:
I think you might have missed part of my comment:

> [...] construct the last path component of the rpath argument using the lowercased form of uname -s - that way Linux, FreeBSD, and any other Unix-like OSes fall into the default case automatically and get something sensible.

`uname -s` will return Linux on Linux, FreeBSD on FreeBSD, etc., thus the lowercased forms of that would be "linux", "freebsd", which should be correct for all non-Apple, non-Windows platforms.